lib/inspec/profile.rb

```diff
@@ -212,6 +212,36 @@ def collect_tests(include_list = @controls)
 
         locked_dependencies.each(&:collect_tests)
 
+        # TODO: if --ludicrous-speed
+        ui = Inspec::UI.new
+        ui.red "*" * 80
+        ui.red "\n\n"
+        ui.red "--ludicrous-speed is an experimental feature that stops\
+ waived controls altogether from being read into InSpec. Use with caution."
+        ui.red "\n\n"
+        ui.red "*" * 80
+
+        ## Find the waivers file
+        collection_source_data_files = @runtime_profile.instance_variable_get(:@src).data_files
+        waiver_path = collection_source_data_files.keys.grep(/waivers.yml$/).first
```

Comment:
So, here's a still-gross-but-a-little-safer way to do this:
```
waiver_path = Inspec::Config.cached.instance_variable_get(:@cli_opts)["waiver_file"]&.first

## Pull together waiver
if waiver_path
  waived_controls = YAML.load_file(waiver_path).keys
end
```

That's better because it reaches less, and it also does not assume that the waiver file will be named `waivers.yml`.

It still has problems:

* We could expose cli_opts and avoid the instance_variable_get
* We are only accessing the first waiver file. What if there are are multiple?
* We don't handle misparsing the waiver file.

Reply:
Reformatted in this way, and manually referenced these issues in a comment above so we don't lose them